### ISSUE TITLE
Add support for Biktorgj's modem firmware

### ISSUE THREAD

**Issue by kescherCode** [kescherCode/kescher-archpp#5]
Biktorgj's modem firmware has a different id in `/dev/serial/by-id/`, therefore, eg25-autorestart needs to be adapted to recognize it as well.

Implementation idea: Replace `device` with `devices` array, and loop over them to find if any matching usb device ids exist.

**Comment by SantX27** [comment]
Usually I open a new issue, but since this already exists, I'll add a comment here.
Today I have installed [Biktorgj's 0.3.1 firmware](https://github.com/Biktorgj/pinephone_modem_sdk/releases/tag/0.3.1) on my modem, then I re-enabled eg25-autorestart, after making sure I got the new `devices` line in the script. After rebooting, the modem dropped every 5 minutes, and it turns out that my modem on 
`/dev/serial/by-id ` doesn't go as `usb-Quectel__Incorporated_LTE_Module_24400821-if0` but as `usb-Quectel__Incorporated_LTE_Module_2c4b4eab-if0`. After adding my modem string on the devices list, the script started working normally. Now I have no idea why the IDs diverge so much (and maybe this should be reported in the pinephone_modem_sdk git), but I think we can generalize Biktorgj's firmware line as usb-Quectel__Incorporated_LTE_Module_*-if0

**Comment by kescherCode** [comment]
Wow, this is an odd difference indeed.
I will add this general check later.

**Comment by ght** [comment]
TL;DR; **after finding the modem under the first device name, the script goes on to check the links under the __other__ device name for existence too, unsurprisingly failing**

---

First of all, thank you @kescherCode  for this tool, it was a big contributing factor to finally daily driving my Pinephone! :)

I updated this morning and was hit by this regression too, so I investigated a bit. Adding some `printf`s to the nested `for` loops checking for the modem and the symlinks quickly showed the problem:
```
Found link, setting modem_found (/dev/serial/by-id/usb-Quectel_EG25-G-if00-port0)
Link missing, setting restart_modem (/dev/serial/by-id/usb-Quectel__Incorporated_LTE_Module_24400821-if00-port0)
```

Should be a quick fix, but if you're busy, I can come up with a PR later today, just let me know.

**Comment by kescherCode** [comment]
Commit a26681560c5eba4f42fb71cd00a476afca8b1178 as well as  eg25-autorestart-0.6.0-1 in the kescher-archpp repo should fix this issue for good. Please report back if the fix works on your devices as well.

**Comment by SantX27** [comment]
Everything is fine on my side, the general check is working. Thanks again!

**Comment by kescherCode** [comment]
I assume this is closed for good now.
